projects/packages/sync/src/modules/class-woocommerce.php

```diff
@@ -242,6 +243,38 @@ public function filter_order_item( $args ) {
 		return $args;
 	}
 
+	/**
+	 * Handler for filtering out non-whitelisted order item meta.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @param array $args Hook arguments.
+	 * @return array|false False if not whitelisted, the original hook args otherwise.
+	 */
+	public function filter_meta( $args ) {
+		if (
+			$this->is_whitelisted_order_item_meta( $args[2] )
+		) {
+			return $args;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Whether an order item meta key is whitelisted for sync.
+	 *
+	 * @access public
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @param string $meta_key Order item meta key.
+	 * @return bool True if whitelisted.
+	 */
+	public function is_whitelisted_order_item_meta( $meta_key ) {
+		return in_array( $meta_key, self::$order_item_meta_whitelist, true );
```

Comment:
Maybe add an extra check for the `$meta_key` here to be on the safe side.
```suggestion
		return is_string( $meta_key ) && in_array( $meta_key, self::$order_item_meta_whitelist, true );
```

Reply:
Good idea. As above though, perhaps we need similar checks in `is_whitelisted_comment_meta` and `is_whitelisted_post_meta` for a follow up 🤔